src/test/java/org/junitpioneer/testkit/assertion/PropertiesAssert.java

```diff
@@ -0,0 +1,205 @@
+	 * {@code getProperty(name)}.  Properties may come from the properties instance itself,
+	 * or from a nested default instance, indiscrimiately.
+	 * <p>
+	 * Properties partially supports object values, but return null for {@code getProperty(name)}
+	 * when the value is a non-string.  This assertion follows the same rules:  Any non-String
+	 * value is considered null for comparison purposes.
+	 *
+	 * @param expected The actual is expected to be effectively the same as this Properties
+	 * @return Assertion instance
+	 */
+	public PropertiesAssert isEffectivelyTheSameAs(Properties expected) {
+
+		// Compare values present in actual
+		actual.propertyNames().asIterator().forEachRemaining(k -> {
+
+			String kStr = k.toString();
+
+			String actValue = actual.getProperty(kStr);
+			String expValue = expected.getProperty(kStr);
+
+			if (actValue == null) {
+				if (expValue != null) {
+					// An object value is the only way to get a null from getProperty()
+					throw failure("For the property '<%s>', "
+							+ "the actual value was an object but the expected the string '<%s>'.",
+						k, expValue);
+				}
+			} else if (!actValue.equals(expValue)) {
+				throw failure("For the property '<%s>', the actual value was <%s> but <%s> was expected", k, actValue,
+					expValue);
+			}
+		});
+
+		// Compare values present in expected - Anything not matching must not have been present in actual
+		expected.propertyNames().asIterator().forEachRemaining(k -> {
+
+			String kStr = k.toString();
+
+			String actValue = actual.getProperty(kStr);
+			String expValue = expected.getProperty(kStr);
+
+			if (expValue == null) {
+				if (actValue != null) {
+
+					// An object value is the only way to get a null from getProperty()
+					throw failure("For the property '<%s>', "
+							+ "the actual value was the string '<%s>', but an object was expected.",
+						k, actValue);
+				}
+			} else if (!expValue.equals(actValue)) {
+				throw failure("The property <%s> was expected to be <%s>, but was missing", k, expValue);
+			}
+		});
+
+		return this;
+	}
+
+	/**
+	 * The converse of isEffectivelyTheSameAs.
+	 *
+	 * @param expected The actual is expected to NOT be effectively the same as this Properties
+	 * @return Assertion instance
+	 */
+	public PropertiesAssert isNotEffectivelyTheSameAs(Properties expected) {
+		try {
+			isEffectivelyTheSameAs(expected);
+		}
+		catch (AssertionError ae) {
+			return this; // Expected
+		}
+
+		throw failure("The actual Properties should not be effectively the same as the expected one.");
+	}
+
+	/**
+	 * Compare values directly present in Properties and recursively into default Properties.
+	 *
+	 * @param expected The actual is expected to be strictly the same as this Properties
+	 * @return Assertion instance
+	 */
+	public PropertiesAssert isStrictlyTheSameAs(Properties expected) {
+
+		// Compare values present in actual
+		actual.keySet().forEach(k -> {
```

Comment:
For this check and below, would it make sense to first collect all failed assertions and then throw the failure?

Reply:
Perhaps.  Other than this is an assertion for the purpose of testing test code.  It could have more features, but I don't think anyone else will ever use it.  It would be a great future improvement if this is ever made public.